packages/slonik/src/service.ts

```diff
@@ -180,9 +230,89 @@ abstract class BaseService<
     return this.factory.table;
   }
 
+  protected getHook(prefix: "pre" | "post", action: string): unknown {
```

Comment:
I would just use `prefix: string`. What if the app dev want to introduce a `doXXX` hook?